app/post.go

```diff
@@ -393,6 +393,14 @@ func (a *App) FillInPostProps(post *model.Post, channel *model.Channel) *model.A
 		post.DelProp("channel_mentions")
 	}
 
+	if !a.HasPermissionToChannel(post.UserId, post.ChannelId, model.PERMISSION_USE_CHANNEL_MENTIONS) {
+		post.AddProp("disable_channel_highlight", true)
+	}
```

Comment:
There is existing logic that handles a prop for channel mentions it is called `POST_PROPS_MENTION_HIGHLIGHT_DISABLED`